### PR TITLE
379 delete rejected prime admin

### PR DESCRIPTION
**Pull request by Sepehr-Sobhani**
[ISSUE 379](https://github.com/orgs/bcgov/projects/122/views/2?pane=issue&itemId=48609618) and [ISSUE 640](https://github.com/orgs/bcgov/projects/122/views/2?pane=issue&itemId=50561029)

### CODE REVIEW

**Comment by andrea-williams** [comment]
Updates based on recent discussion with Hanna:

- IRC will want to still see rejected prime admin requests, in case external users contact them to follow up on access requests
- the "Senior Officer" part of the form will be removed as part of MVP (IRC no longer needs it) - this might mean this PR is no longer needed? I haven't read through the code for this PR yet - let me know if I should @Sepehr-Sobhani !

**Comment by Sepehr-Sobhani** [comment]
> Updates based on recent discussion with Hanna:
> 
> * IRC will want to still see rejected prime admin requests, in case external users contact them to follow up on access requests
> * the "Senior Officer" part of the form will be removed as part of MVP (IRC no longer needs it) - this might mean this PR is no longer needed? I haven't read through the code for this PR yet - let me know if I should @Sepehr-Sobhani !

@andrea-williams 
- Thanks for confirming.
- I think this PR is still valid because we are refactoring a bunch of code here but we need a separate task ticket to remove the SO part from the code.

File: bc_obps/registration/api/user_operator.py [hunk 1]
```diff
@@ -358,71 +358,38 @@ def create_user_operator_contact(request, payload: UserOperatorContactIn):
...
+    if payload.user_guid:  # to update the status of a user_operator by user_guid
+        user_operator = get_object_or_404(UserOperator, user_id=payload.user_guid)
+    elif payload.user_operator_id:  # to update the status of a user_operator by user_operator_id
+        user_operator = get_object_or_404(UserOperator, id=payload.user_operator_id)
+    else:
+        return 404, {"message": "No parameters provided"}
+
+    # We can't update the status of a user_operator if the operator is not approved
+    if user_operator.operator.status != Operator.Statuses.APPROVED:
+        return 400, {"message": "Operator must be approved before approving users."}
```
Comment (andrea-williams) [inline bc_obps/registration/api/user_operator.py:374]: The logic here will need to be modified. In cases where the operator data has been imported from CIIP/SWRS, its status will be "DRAFT" at the time that a user is requesting prime admin. The user will need to have admin access to the operator before they're able to update/confirm the imported data about the Operator, at which point the Operator's status will be set to Pending/Accepted.

(see the "Andrea's brain dumping ground" Miro board for a map of this workflow if you need more context)

**Review by andrea-williams — CHANGES_REQUESTED** [review]
Just the one change request (related to my comment in `user_operator.py`). I think the fix is as simple as changing
`if user_operator.operator.status != Operator.Statuses.APPROVED:`
to

`if user_operator.operator.status is in [Operator.Statuses.PENDING, Operator.Statuses.REJECTED]:`

**Comment by Sepehr-Sobhani** [comment]
> Just the one change request (related to my comment in `user_operator.py`). I think the fix is as simple as changing `if user_operator.operator.status != Operator.Statuses.APPROVED:` to
> 
> `if user_operator.operator.status is in [Operator.Statuses.PENDING, Operator.Statuses.REJECTED]:`

I thought users always have to approve the operator first and then they will be prompted to approve the admin request!

**Comment by andrea-williams** [comment]
> > Just the one change request (related to my comment in `user_operator.py`). I think the fix is as simple as changing `if user_operator.operator.status != Operator.Statuses.APPROVED:` to
> > `if user_operator.operator.status is in [Operator.Statuses.PENDING, Operator.Statuses.REJECTED]:`
> 
> I thought users always have to approve the operator first and then they will be prompted to approve the admin request!

When it's a newly created operator (i.e., one that hasn't been imported from CIIP/SWRS), an internal user will have to approve the new operator first before approving the admin request. But when it's an operator that has been imported from existing data, we need to approve the admin request first before we allow that external user to review the imported Operator data and update any fields that have changed.

I hope this makes sense - if not, happy to hop on a call to discuss it :)

**Comment by Sepehr-Sobhani** [comment]
> > > Just the one change request (related to my comment in `user_operator.py`). I think the fix is as simple as changing `if user_operator.operator.status != Operator.Statuses.APPROVED:` to
> > > `if user_operator.operator.status is in [Operator.Statuses.PENDING, Operator.Statuses.REJECTED]:`
> > 
> > 
> > I thought users always have to approve the operator first and then they will be prompted to approve the admin request!
> 
> When it's a newly created operator (i.e., one that hasn't been imported from CIIP/SWRS), an internal user will have to approve the new operator first before approving the admin request. But when it's an operator that has been imported from existing data, we need to approve the admin request first before we allow that external user to review the imported Operator data and update any fields that have changed.
> 
> I hope this makes sense - if not, happy to hop on a call to discuss it :)

Gotcha, Thanks for the clarification.

File: bc_obps/registration/api/user_operator.py [hunk 2]
```diff
@@ -358,71 +358,38 @@ def create_user_operator_contact(request, payload: UserOperatorContactIn):
...
+def update_user_operator_status(request, payload: UserOperatorStatusUpdate):
+    current_user: User = request.current_user  # irc user or industry user admin
+    if payload.user_guid:  # to update the status of a user_operator by user_guid
+        user_operator = get_object_or_404(UserOperator, user_id=payload.user_guid)
+    elif payload.user_operator_id:  # to update the status of a user_operator by user_operator_id
+        user_operator = get_object_or_404(UserOperator, id=payload.user_operator_id)
+    else:
+        return 404, {"message": "No parameters provided"}
+
+    # We can't update the status of a user_operator if the operator is not approved
```
Comment (andrea-williams) [inline bc_obps/registration/api/user_operator.py:372]:
```suggestion
    # We can't update the status of a user_operator if the operator has been declined or is awaiting review
```

**Pull request merged**